File: controlePedidos/src/main/java/com/br/chocolatePontoVirgula/model/entity/Produto.java
package com.br.chocolatePontoVirgula.model.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.sql.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "produto")
public class Produto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)

    @Column(name = "id")
    private Long id;
    @Column(name = "nome")
    private String nome;
    @Column(name = "descricao")
    private String descricaoProduto;
    @Column(name = "unidade_medida")
    private String unidadeMedida;
    @Column(name = "valor_unitario")
    private double valorUnitario;
    @Column(name = "status")
    private boolean status;
    @Column(name = "url_Fotos")
    private String urlFoto;
    @Column(name = "peso")
    private double peso;
    @Column(name = "data_fabricacao")
    private Date dataFabricacao;
    @Column(name = "data_validade")
    private Date dataValidade;
    @Column(name = "quantidade_estoque")
    private Integer quantidadeEstoque;


    public String Status() {
        if (status) {
            return "Ativo";
        } else {
            return "Inativo";
        }
    }


    public void setStatus(boolean status) {
        this.status = status;
    }

}

File: controlePedidosFront/src/app/shared/models/produto.ts
export class Produto {
  id: number;
  nome?: string;
  descricaoProduto?: string;
  unidadeMedida?: string;
  valorUnitario?: number;
  status?: boolean;
  urlFoto?: string;
  peso?: number;
  dataFabricacao?: Date;
  dataValidade?:Date;
  quantidadeEstoque?: number;

  constructor(id: number, nome: string, descricao: string, unidade_medida: string,
     valor_unitario: number, status: boolean,urlFoto: string, peso: number,
     data_fabricacao: Date, data_validade: Date, quantidade_estoque: number ){
    this.id = id;
    this.nome = nome;
    this.descricaoProduto = descricao;
    this.unidadeMedida = unidade_medida;
    this.valorUnitario = valor_unitario;
    this.status = status;
    this.urlFoto = urlFoto;
    this.peso = peso;
    this.dataFabricacao = data_fabricacao;
    this.dataValidade = data_validade;
    this.quantidadeEstoque = quantidade_estoque;
  }
}


